RiotSwiftUI/Modules/Room/PollHistory/View/PollHistory.swift

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct PollHistory: View {
+    @Environment(\.theme) private var theme
+    
+    @ObservedObject var viewModel: PollHistoryViewModel.Context
+    
+    var bindings: PollHistoryViewBindings {
+        viewModel.viewState.bindings
+    }
+    
+    var body: some View {
+        VStack {
+            HStack {
+                SegmentedPicker(
+                    segments: PollHistoryMode.allCases.map { ($0.segmentTitle, $0) },
+                    selection: $viewModel.mode,
+                    interSegmentSpacing: 14
+                )
+                Spacer()
+            }
+            .padding(.horizontal, 16)
+            
+            if viewModel.viewState.polls.isEmpty {
+                noPollsView
+            } else {
+                pollListView
+            }
+        }
+        .padding(.top, 32)
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(theme.colors.background.ignoresSafeArea())
+        .accentColor(theme.colors.accent)
+        .navigationTitle(VectorL10n.pollHistoryTitle)
+        .onAppear {
+            viewModel.send(viewAction: .viewAppeared)
+        }
+    }
+    
+    private var pollListView: some View {
+        ScrollView {
+            LazyVStack(spacing: 32) {
+                let enumeratedPolls = Array(viewModel.viewState.polls.enumerated())
+                
+                ForEach(enumeratedPolls, id: \.offset) { _, pollData in
```

Comment:
It seems more sensible to make `PollListData` conform to `Identifiable` (using the poll start event ID presumably) than enumerating them just for an index which is potentially an unstable ID for the poll if the order changed.

Reply:
`PollListData` is totally a mock for now.
This will change a lot in next PRs I guess...